solidity/contracts/libraries/operator/GroupSelection.sol

```diff
@@ -94,6 +97,17 @@ library GroupSelection {
         self.inProgress = true;
         self.seed = _seed;
         self.ticketSubmissionStartBlock = block.number;
+
+        // Calculate possible operators count by dividing total token supply
+        // by current minimum stake.
+        uint256 possibleOperatorsCount = uint256(10**27).div(self.minimumStake);
+
+        // ticketSubmissionTimeout consists of:
+        // - one initial ticket submission round
+        // - N reactive ticket submission rounds
+        // - one additional waiting round
+        self.ticketSubmissionTimeout = self.ticketSubmissionRoundDuration
+            .mul(reactiveTicketSubmissionRounds(possibleOperatorsCount, self.groupSize).add(2));
```

Comment:
Worth considering putting this code in a function. `internal view` functions are cheap, e.g.
```
function calculateTicketSubmissionTimeout() internal view returns (uint256) {
  // Calculate possible operators count by dividing total token supply
  // by current minimum stake.
  uint256 possibleOperatorsCount = uint256(tokenSupply).div(self.minimumStake);
  
  // ticketSubmissionTimeout consists of:
  // - one initial ticket submission round
  // - N reactive ticket submission rounds
  // - one additional waiting round
  uint256 reactiveRounds = reactiveTicketSubmissionRounds(
    possibleOperatorsCount, 
    self.groupSize
  )
  return self.ticketSubmissionRoundDuration.mul(reactiveRounds.add(2));
}
```

Reply:
Might as well make all of this `constant`. We also need to add the mining lag at the end, to ensure that submissions will be reliably mined even in the worst-case scenario.

Reply:
Not applicable now as we removed the dynamic submission timeout.